Display the number of items that would be deleted when replacing the online library

tnajdek opened the pull request:
This compares item keys in the remote library with those in the local DB and prints an additional warning if at least one remote key is not found locally. In a fully synced library, there is no change in behaviour, but if I remove `zotero.sqlite`, I’ll now see:

![Screenshot 2025-06-26 at 14 24 46](https://github.com/user-attachments/assets/213a2797-e338-4d3a-a8fd-97bf3ba136ee)

Few notes:

* I've added `asKeys` to `getAll` in `items.js` as an optimisation. If we don't want to affect the core API for this, I can retrieve all items and map them to keys instead.
* We probably need a spinner in the Preferences window to provide feedback while remote keys are being fetched.

Fix #5078

dstillman commented:
> I've added `asKeys` to `getAll` in `items.js` as an optimisation. If we don't want to affect the core API for this, I can retrieve all items and map them to keys instead.

Yeah, I would just use `asIDs` and map with `Zotero.Items.getLibraryAndKeyFromID(id)`.

dstillman commented on chrome/locale/en-US/zotero/preferences.ftl:
```diff
@@ -99,3 +99,12 @@ preferences-advanced-custom-data-dir =
+        [1] item
+        *[other] items
+    } in the online library will be deleted!
+}
```
Probably better to do this as a separate string that gets added after two newlines if >0.

And I think we can omit the exclamation mark. Arguably appropriate for thousands of items, but it might say "1 item", and in general it's probably better to be matter-of-fact about it.

dstillman commented on chrome/content/zotero/preferences/preferences_sync.jsx:
```diff
@@ -831,21 +831,24 @@ Zotero_Preferences.Sync = {
+				var keyInfo = await Zotero.Sync.Runner.checkAccess(client, { timeout: 5000 });
+				let { keys: remoteKeysArray } = await client.getKeys('user', keyInfo.userID, { target: 'items/top' });
+				let remoteKeys = new Set(remoteKeysArray);
+				let localKeys = new Set(await Zotero.Items.getAll(Zotero.Libraries.userLibraryID, true, false, false, true));
```
This is using `onlyTopLevel`, so in theory you could be wiping out hundreds of attachments, notes, and/or annotations. Probably too confusing to count annotations as "items", but we should probably at least count attachments and notes. Can just omit `asIDs` (contra my other comment), filter to `isRegularItem() || isNote() || isFileAttachment()`, and map to`.key`.

We could also show annotations separately, but that's probably getting too complicated, since the main use case here is just preventing wiping out a whole library.

tnajdek commented:
> > I've added `asKeys` to `getAll` in `items.js` as an optimisation. If we don't want to affect the core API for this, I can retrieve all items and map them to keys instead.
> 
> Yeah, I would just use `asIDs` and map with `Zotero.Items.getLibraryAndKeyFromID(id)`.

I've reverted this change

dstillman commented:
Great, thanks!
